### PR TITLE
Add BackboneMixin

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Add `BackboneMixin` with a method `forward_with_filtered_kwargs`.

### CODE REVIEW

File: src/transformers/models/bit/modeling_bit.py [hunk 1]
```diff
@@ -842,7 +842,7 @@ def forward(
     """,
     BIT_START_DOCSTRING,
 )
-class BitBackbone(BitPreTrainedModel):
+class BitBackbone(BitPreTrainedModel, BackboneBaseModel):
```
Comment (NielsRogge) [inline src/transformers/models/bit/modeling_bit.py:845]: It's not really convenient that any backbone will need to inherit from 2 classes can we directly incorporate this into PreTrainedModel or not?

Comment (ydshieh) [reply src/transformers/models/bit/modeling_bit.py:845]: SGTM, but just let @sgugger to confirm

Comment (sgugger) [reply src/transformers/models/bit/modeling_bit.py:845]: I don't see where the problem lies. That's the most Pythonic way of doing this (compared to having all models inherit from `GenerationMixin` for instance, whereas lots of them shouldn't). You give the type to objects that need it (and only those).

**Review by sgugger — APPROVED** [review]
Thanks for working on this @ydshieh !

**Pull request merged**